src/ax/marshaling/json.go

```diff
@@ -0,0 +1,108 @@
+package marshaling
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"mime"
+	"reflect"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
+)
+
+const (
+	// JSONContentType is the content-type name for JSON-encoded messages.
+	JSONContentType = "application/json"
+)
+
+// MarshalJSON marshals a ProtocolBuffers messages to JSON
+// representation and returns a MIME content-type that identifies the particular
+// message protocol.
+func MarshalJSON(msg proto.Message) (ct string, data []byte, err error) {
```

Comment:
Slight grammar issue in docs, perhaps say `... marshals a ProtocolBuffers message to its JSON representation ...`